include/aos/iam/identhandler.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_IDENTHANDLER_HPP_
+#define AOS_IDENTHANDLER_HPP_
+
+#include "aos/common/types.hpp"
+#include "aos/iam/config.hpp"
+
+namespace aos {
+namespace iam {
+namespace identhandler {
+
+/** @addtogroup iam Identification and Access Manager
+ *  @{
+ */
+
+/**
+ * Subjects changed observer interfaceю
+ */
+class SubjectsChangedObserverItf {
+public:
+    /**
+     * Subjects Changed Observer interface.
+     *
+     * @param[in] messages subject changed messages.
+     * @returns Error.
+     */
+    virtual Error SubjectsChanged(const Array<StaticString<cSubjectIDLen>>& messages) = 0;
+
+    /**
+     * Destroys subjects changed observer interface.
+     */
+    virtual ~SubjectsChangedObserverItf() = default;
+};
```

Comment:
It dont know how it is going to be utilized, but it loooks like an ordinary functor,
Probably it would be better to implement a Function object for the same usecases.

Reply:
Let's get back to this question on the actual IdentHandlerItf implementation PR

Reply:
ок